controller/Security.php

```diff
@@ -80,6 +85,21 @@ public function oidcInitiation(): void
         $this->redirect($message->toUrl());
     }
 
+    private function getKeyChainRepository(): KeyChainRepositoryInterface
+    {
+        return $this->getServiceLocator()->get(CachedPlatformKeyChainRepository::class);
+    }
+
+    private function getPlatformKeyChainRepository(): PlatformKeyChainRepository
+    {
+        return $this->getServiceLocator()->get(PlatformKeyChainRepository::SERVICE_ID);
+    }
+
+    private function getAuthorizationServerFactory(): AuthorizationServerFactory
+    {
+        return $this->getServiceLocator()->getContainer()->get(AuthorizationServerFactory::class);
```

Comment:
I'm pretty sure these can be unified by accessing a service by its FQN.
```suggestion
        return $this->getServiceLocator()->get(PlatformKeyChainRepository::class);
    }

    private function getAuthorizationServerFactory(): AuthorizationServerFactory
    {
        return $this->getServiceLocator()->get(AuthorizationServerFactory::class);
```

Reply:
Perhaps, additional changes will be required in the container mock configuration as well in some of the tests.